### DISCUSSION TITLE
ECDHE ciphers for iPXE

### DISCUSSION THREAD

**Discussion by sgrigorov** [ipxe/ipxe#1135] [Ideas]
Hi All,

Does anyone know what is going to take to have the ECDHE ciphers implemented in iPXE?
There are some mentions during the years but it is still not implemented. 

Thanks

https://lists.ipxe.org/pipermail/ipxe-devel/2020-June/007079.html
https://forum.ipxe.org/showthread.php?tid=11985&highlight=ECDHE
https://ipxe-devel.ipxe.narkive.com/dHkhXzdL/ecdhe-rsa-cipher-suites

**Comment by NiKiZe** [comment]
Some input #1101, #1102, #787

Especially https://github.com/ipxe/ipxe/issues/787#issuecomment-1294852530

**Reply by sgrigorov** [reply]
Any idea what funding is required to get this implemented?

**Comment by mcb30** [comment]
Impeccable timing to ask this question :upside_down_face: 

The final commit for ECDHE and x25519 support was pushed just now: https://github.com/ipxe/ipxe/commit/963ec1c4f
